h/views/oidc.py

```diff
@@ -422,16 +427,14 @@ def log_in_with_provider(self, provider_unique_id: str, user):
                         provider_unique_id,
                         self.settings.jwt_issuer,
                         self.settings.idinfo_jwtaudience,
+                        next_url,
                     ),
                 )
             )
 
         headers = login(user, self._request)
 
-        return HTTPFound(
-            self._request.route_url("activity.user_search", username=user.username),
-            headers=headers,
-        )
+        return HTTPFound(redirect_url(self._request, user, next_url), headers=headers)
```

Comment:
This redirects to the `next` URL after successfully logging in to a provider account that already existed.